### PR TITLE
Add get_feature_pipelines

### PR DESCRIPTION
**Pull request by tracivar**
- Returned `feature_pipelines` is a list of pipelines that are ready to be used for `add_feature_pipeline`
- Amended `parse_source` to accept source name

https://zegami.atlassian.net/browse/ZGM-8263

### CODE REVIEW

File: zegami_sdk/collection.py [hunk 1]
```diff
@@ -367,6 +370,88 @@ def add_feature_pipeline(self, pipeline_name, steps, source=0, generate_snapshot
...
+            feature_pipelines_nodes[source_name] = {}
+
+        for node in all_nodes:
+            node_groups = node.get('node_groups')
+            # check if node_groups contain 'feature_pipeline_'
+            if node_groups and len(node_groups) == 3:
+                # get the source name after 'source_'
+                node_source_name = node_groups[0][7:]
+                # get the source name after 'feature_pipeline_'
+                feature_pipeline_name = node_groups[2][17:]
```
Comment (ahaith) [inline zegami_sdk/collection.py:425]: Probably not safe to have these assumptions about the orders of the node groups.
Better to do something like
```
[group for group in node_groups if node_group.startswith('feature_pipeline')][0]
```

File: zegami_sdk/collection.py [hunk 2]
```diff
@@ -367,6 +370,88 @@ def add_feature_pipeline(self, pipeline_name, steps, source=0, generate_snapshot
...
+                        {
+                            'action': 'mRMR',
+                            'params': mRMR_params,
+                        },
+                        {
+                            'action': 'cluster',
+                            'params': cluster_params,
+                        }
+                    ]
+                })
```
Comment (ahaith) [inline zegami_sdk/collection.py:452]: This makes a hard assumption that it's always mRMR > cluster. It'd be a bit more future-proof to just map nodes to steps, (assuming we reliably get the nodes in the correct order):
```
steps = [make_step(node) for node in nodes]
```

**Review by ahaith — APPROVED** [review]
Looks good. A couple of suggestions to make this more robust.
Well worth adding an e2e test for this.

File: zegami_sdk/collection.py (hunk 2 above)
Comment (tracivar) [reply zegami_sdk/collection.py:452]: But nodes are not in correct order as we get them from the node statuses, and there is a mapping node that we won't need to use

**Pull request merged**